substrate/client/network/src/litep2p/shim/notification/tests/peerset.rs

```diff
@@ -1123,3 +1126,174 @@ async fn set_reserved_peers_cannot_move_previously_reserved() {
 	assert_eq!(peerset.num_out(), 0usize);
 	assert_eq!(peerset.reserved_peers().len(), 3usize);
 }
+
+#[tokio::test]
+async fn reserved_only_rejects_non_reserved_peers() {
```

Comment:
dq: is there a place where can I find more about reserved peers? Not sure what they are.

Reply:
Upsy, there is a dedicated PR for this I think https://github.com/paritytech/polkadot-sdk/pull/8650. Will try to follow it, resolving this.